### PR TITLE
Switched to Algolia for searching

### LINKED ISSUE #3: Consider using Algolia for search
Hey, I work for Algolia and saw this project. It would be nice for features like typo-tolerance to simply add a script to push to Algolia and use for example React InstantSearch to display and filter the results. 

This will most probably fit under our free plan, and if it wouldn't, then it'd fit under the open source plan. 

Happy to help setting it up!

### PR DESCRIPTION
**Pull request by gantoine**
Closes #3

### CODE REVIEW

File: src/utils/api.js [hunk 1]
```diff
@@ -37,10 +42,28 @@ function navigateDefinition(directory): NetResult[] {
     .value();
 };
 
+const algoliaSearch = _.debounce((params, cb) => {
```
Comment (Haroenv) [inline src/utils/api.js]: no need to debounce, if you hit into QPS limits, we'll make sure you get enough to make it work.

**Review by Haroenv — COMMENTED** [review]
Do you need anything (process how to index maybe) to make this work?

**Comment by gantoine** [comment]
I've sorta put this on the back-burner. Indexing will mean dealing with webhooks on the real flow-typed repo, and I like to keep this project self-contained. Currently the only limit is the GitHub API request limit, which I don't expect anyone to hit.

I can't see the benefits of Algolia outweighing the extra work (for now), but I'm going to keep the PR open, as I'll want to revisit it sometime in the near future.

**Comment by Haroenv** [comment]
All right, seems fair

**Comment by gantoine** [comment]
@Haroenv Good news! I (finally) got around to doing daily indexing of definitions in Algolia. I have another project up on Heroku that fetches and indexes the definitions daily. The latest version of `search-flow-typed` is now using it to search!

**Comment by Haroenv** [comment]
Nice!

**Pull request merged**